src/prefect/cli/deploy.py

```diff
@@ -1151,6 +1161,62 @@ def _filter_matching_deploy_config(name, deploy_configs):
     return matching_deployments
 
 
+def _parse_name_from_pattern(deploy_configs, name_pattern):
+    """
+    Parse the deployment names from a user-provided pattern such as "flow-name/*" or "my-deployment-*"
+
+    Example:
+
+    >>> deploy_configs = [
+    ...     {"name": "my-deployment-1", "entrypoint": "flow-name-1"},
+    ...     {"name": "my-deployment-2", "entrypoint": "flow-name-2"},
+    ...     {"name": "my-deployment-3", "entrypoint": "flow-name-3"},
+    ... ]
+
+    >>> _parse_name_from_pattern(deploy_configs, "flow-name-1/*")
+    ["my-deployment-1"]
+
+    Args:
+        deploy_configs: A list of deploy configs
+        name: A pattern to match against the deploy configs
+
+    Returns:
+        List[str]: a list of deployment names that match the given pattern
+    """
+    parsed_names = []
+
+    # eg. "flow-name/deployment-name"
+    if "/" in name_pattern:
```

Comment:
I recommend that you start by regex-escaping the string, just in case the user is trying to match on characters that have special meaning in regex (like 'show-me-the-$/*', where that `$` is a reserved character in regex).  When you escape the regex first, then all of the `*` the user had in their pattern will be turned into `\*`, so you can then replace them back with the `.*`:

```suggestion
    parsed_names = []
    
    name_pattern = re.escape(name_pattern).replace(r"\*", ".*")

    # eg. "flow-name/deployment-name"
    if "/" in name_pattern:
```

Reply:
Ah, yeah another great call! I didn't realize users could add a "$" but looking at our [banned chars  for deployment and flow names](https://github.com/PrefectHQ/prefect/blob/653e6217d1728ceb06f5273945c21b29eb0eac7e/src/prefect/_internal/schemas/validators.py#L10), that seems to be the case.